pages/_document.js

```diff
@@ -1,31 +1,36 @@
 import Document, { Html, Head, Main, NextScript } from 'next/document';
 import { CssBaseline } from '@geist-ui/react';
+import { ServerStyleSheet } from 'styled-components';
 
 class MyDocument extends Document {
   static async getInitialProps(ctx) {
-    const initialProps = await Document.getInitialProps(ctx);
+    // for geist-ui CSS baseline
     const styles = CssBaseline.flush();
 
-    return {
-      ...initialProps,
-      styles: (
-        <>
-          {initialProps.styles}
-          {styles}
-        </>
-      ),
-    };
-  }
+    // for styled-components
+    const sheet = new ServerStyleSheet();
+    const originalRenderPage = ctx.renderPage;
+    try {
+      ctx.renderPage = () =>
+        originalRenderPage({
+          enhanceApp: (App) => (props) =>
+            sheet.collectStyles(<App {...props} />),
+        });
 
```

Comment:
@mattvv Maybe you don't see the loading issue on this PR because of this. When both PR's get merge we can check if the problem continues. If it does I can work on a fix.

Reply:
You're right @RobertoValle31 - looks like @shc023 fixed the server side rendering for styled components- nice find